include/ensmallen_bits/problems/logistic_regression_function_impl.hpp

```diff
@@ -64,19 +66,22 @@ LogisticRegressionFunction<MatType>::LogisticRegressionFunction(
 /**
  * Shuffle the datapoints.
  */
-template<typename MatType>
-void LogisticRegressionFunction<MatType>::Shuffle()
+template<typename MatType, typename LabelsType>
+void LogisticRegressionFunction<MatType, LabelsType>::Shuffle()
 {
   MatType newPredictors;
-  arma::Row<size_t> newResponses;
+  LabelsType newResponses;
 
   arma::uvec ordering = arma::shuffle(arma::linspace<arma::uvec>(0,
       predictors.n_cols - 1, predictors.n_cols));
 
   newPredictors.set_size(predictors.n_rows, predictors.n_cols);
+  newResponses.set_size(responses.n_elem);
   for (size_t i = 0; i < predictors.n_cols; ++i)
+  {
     newPredictors.col(i) = predictors.col(ordering[i]);
-  newResponses = responses.cols(ordering);
```

Comment:
If you make `ordering` into `LabelsType` (or some equivalent unsigned vector type), then I think `.cols()` should work on the GPU too?